app/projectwizard.cpp

```diff
@@ -0,0 +1,191 @@
+#include "projectwizard.h"
+#include "inpututils.h"
+
+#include "qgsproject.h"
+#include "qgsrasterlayer.h"
+#include "qgsvectorlayer.h"
+#include "qgsvectorfilewriter.h"
+#include "qgsdatetimefieldformatter.h"
+
+ProjectWizard::ProjectWizard( const QString &dataDir, FieldsModel *fieldsModel, QObject *parent )
+  : QObject( parent )
+  , mDataDir( dataDir )
+  , mFieldsModel( fieldsModel )
+{
+  QObject::connect( mFieldsModel, &FieldsModel::notify, this, &ProjectWizard::notify );
+}
+
+QgsVectorLayer *ProjectWizard::createGpkgLayer( QString const &projectDir )
+{
+  QString gpkgName( QStringLiteral( "data" ) );
+  QString projectGpkgPath( QString( "%1/%2.%3" ).arg( projectDir ).arg( gpkgName ).arg( "gpkg" ) );
+  QString layerName( QStringLiteral( "Points" ) );
+  QgsCoordinateReferenceSystem layerCrs( "EPSG:4326" );
+  QgsFields predefinedFields = createFields( mFieldsModel->fields() );
+
+  // Write layer as gpkg
+  QgsVectorLayer *layer = new QgsVectorLayer( QStringLiteral( "Point?crs=EPSG:4326" ), layerName, "memory" );
+  layer->startEditing();
+  layer->setCrs( layerCrs );
+  for ( QgsField f : predefinedFields )
+  {
+    layer->addAttribute( f );
+  }
+  layer->updateFields();
+  layer->commitChanges();
+
+  QgsVectorFileWriter::SaveVectorOptions options;
+  options.driverName = QStringLiteral( "GPKG" );
+  options.layerName = layerName;
+  options.fileEncoding = QStringLiteral( "UTF-8" );
+  std::unique_ptr< QgsVectorFileWriter > writer( QgsVectorFileWriter::create( projectGpkgPath, predefinedFields, QgsWkbTypes::Point, layerCrs, QgsCoordinateTransformContext(), options ) );
+
+
+  QString errorMessage;
+  QgsVectorFileWriter::writeAsVectorFormatV2(
+    layer,
+    projectGpkgPath,
+    layer->transformContext(),
+    options,
+    nullptr,
+    nullptr,
+    &errorMessage );
+
+  // Check and configure layer
+  QgsVectorLayer *l = new QgsVectorLayer( projectGpkgPath, "Points", "ogr" );
+
+  Q_ASSERT( l->isValid() );
+
+  l->startEditing();
+  l->setCrs( layerCrs );
+  for ( int i = 0; i < l->fields().count(); ++i )
+  {
+    QgsField f = l->fields().at( i );
+    QgsEditorWidgetSetup setup = getEditorWidget( f, mFieldsModel->findWidgetTypeByFieldName( f.name() ) );
+    l->setEditorWidgetSetup( i, setup );
+  }
+  l->commitChanges();
+
+  return l;
+}
+
+void ProjectWizard::createProject( QString const &projectName )
+{
```

Comment:
Lets check for storage permission first, or better, ask for it when user clicks on "Create Project"

```suggestion
{
if .. InputUtils::hasLocationPermission() 
```